### PR TITLE
add service account creation and annotations

### PR DESCRIPTION
**Pull request by bruce-y**
This allows users to configure annotations against their OPA workloads to leverage authentication methods such as kube2iam or service account token volume projection.

### CODE REVIEW

File: helm/templates/deployment-controller.yaml [hunk 1]
```diff
@@ -20,15 +20,15 @@ spec:
       maxSurge: 100%
   template:
     metadata:
-      {{- with .Values.podAnnotations }}
+      {{- with .Values.controller.podAnnotations }}
```
Comment (taha-au) [inline helm/templates/deployment-controller.yaml:23]: let's have both common annotations and labels, as well as specific ones.

File: helm/values.yaml [hunk 2]
```diff
@@ -95,6 +95,22 @@ controller:
   extraEnv: {}
   envFrom: []
 
+  serviceAccount:
+    create: true
+    name: superblocks-controller
```
Comment (taha-au) [inline helm/values.yaml:100]: nit: i think this is returned by a helper

File: helm/values.yaml [hunk 3]
```diff
@@ -95,6 +95,22 @@ controller:
   extraEnv: {}
   envFrom: []
 
+  serviceAccount:
+    create: true
```
Comment (taha-au) [inline helm/values.yaml:99]: do we want to create by default?

File: helm/values.yaml (hunk 2 above)
Comment (bruce-y) [reply helm/values.yaml:100]: can't use helpers in the values file right?

File: helm/values.yaml (hunk 3 above)
Comment (bruce-y) [reply helm/values.yaml:99]: yes, I think this is desirable. it's mostly standard practice to not use the default service account when releasing a chart

**Review by taha-au — APPROVED** [review]

**Pull request merged**